sim/core/item_swaps.go

```diff
@@ -284,18 +308,23 @@ func (swap *ItemSwap) EligibleSlotsForItem(itemID int32) []proto.ItemSlot {
 
 	if !swap.IsEnabled() {
 		return eligibleSlots
-	} else {
-		return FilterSlice(eligibleSlots, func(slot proto.ItemSlot) bool {
-			return (swap.originalEquip[slot].ID == itemID) || (swap.swapEquip[slot].ID == itemID)
-		})
 	}
+
+	return FilterSlice(eligibleSlots, func(slot proto.ItemSlot) bool {
+		return (swap.originalEquip[slot].ID == itemID) || (swap.swapEquip[slot].ID == itemID)
+	})
+
 }
 
 func (swap *ItemSwap) EligibleSlotsForEffect(effectID int32) []proto.ItemSlot {
 	var eligibleSlots []proto.ItemSlot
 
-	if swap.IsEnabled() {
-		for itemSlot := proto.ItemSlot(0); itemSlot < NumItemSlots; itemSlot++ {
+	for itemSlot := proto.ItemSlot(0); itemSlot < NumItemSlots; itemSlot++ {
+		if !swap.IsEnabled() {
+			if swap.character.Equipment.containsEnchantInSlot(effectID, itemSlot) {
+				eligibleSlots = append(eligibleSlots, itemSlot)
+			}
```

Comment:
I think this will crash if `player.EnableItemSwap` is `false` in the sim request, since `swap.character` will be `nil` in that scenario. For example, I would expect a Feral sim with Landslide equipped to crash on this branch.

Reply:
Oh never mind, `swap.character` will always be set during `initialize()` even if item swapping was not configured, so this should be fine.

Reply:
Yea I made sure to test this before